### DISCUSSION TITLE
WLAN hostname works with Windows clients, fails with iPhone and iPad

### DISCUSSION THREAD

**Discussion by Quadrangle** [micropython/micropython#15842] [General]
We created a wireless web server using Micropython v1.23 in an ESP32, using the WLAN class.  For STA connections, we implemented hostname, allowing users to enter into browsers: “http://hostname”.  Works beautifully with Windows PCs, but never on any iOS device.  We went pretty deep into Apple tech support and they had no clue.

We’re thinking of implementing a static IP to work around this.  

Any thoughts, hints or suggested will be appreciated!

**Comment by Josverl** [comment]
Have you looked into mdns names?
I think apple was one of the first to use that protocol, although by an other name , 'Bonjour'

That would be `http://hostname.local`

**Reply by Quadrangle** [reply]
THANK YOU, IT WORKED!!!!!!!!!!!!!!!!!!!!!

I owe you big time.
I am summoning the universe to bring you happiness, peace and prosperity.
